Common/Securities/Crypto/Crypto.cs

```diff
@@ -0,0 +1,105 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using QuantConnect.Data;
+using QuantConnect.Orders.Fees;
+using QuantConnect.Orders.Fills;
+using QuantConnect.Orders.Slippage;
+using QuantConnect.Securities.Forex;
+
+namespace QuantConnect.Securities.Crypto
+{
+    /// <summary>
+    /// Crypto Security Object Implementation for Crypto Assets
+    /// </summary>
+    /// <seealso cref="Security"/>
+    public class Crypto : Security, IBaseCurrencySymbol
+    {
+        /// <summary>
+        /// Constructor for the Crypto security
+        /// </summary>
+        /// <param name="exchangeHours">Defines the hours this exchange is open</param>
+        /// <param name="quoteCurrency">The cash object that represent the quote currency</param>
+        /// <param name="config">The subscription configuration for this security</param>
+        /// <param name="symbolProperties">The symbol properties for this security</param>
+        public Crypto(SecurityExchangeHours exchangeHours, Cash quoteCurrency, SubscriptionDataConfig config, SymbolProperties symbolProperties)
+            : base(config,
+                quoteCurrency,
+                symbolProperties,
+                new CryptoExchange(exchangeHours),
+                new ForexCache(),
+                new SecurityPortfolioModel(),
+                new ImmediateFillModel(),
+                new InteractiveBrokersFeeModel(),
```

Comment:
Perhaps constant fee model (0)? I thought most exchanges were spread based?

Reply:
The gdax model is fee-free for makers after rebates. The other exchanges I have in mind, OKCoin, Bitfinex and Kraken have a maker/taker split at around 0.01%/0.02%. The Chinese exchanges used to offer fee free until the PBOC "visits" earlier this year. It makes sense to provide a constant fee model with the expectation that it needs overriding as anyone writing an algorithm for crypto that does not specify a valid fee model will get uselessly biased results.